SpeechSandbox/Assets/Scripts/SpeechSandboxStreaming.cs

```diff
@@ -295,83 +285,87 @@ private void OnRecognize(SpeechRecognitionEvent result, Dictionary<string, objec
+                            Log.Debug("SpeechSandboxStreaming.OnRecognize()", "\t word: {0} | confidence: {1}", alternative.word, alternative.confidence);
                     }
                 }
             }
         }
     }
 
-    void OnMessage(object resp, Dictionary<string, object> customData)
+    private void OnMessage0(DetailedResponse<MessageResponse> resp, IBMError error)
     {
-        //  Convert resp to fsdata
-
-        fsData fsdata = null;
-        fsResult r = _serializer.TrySerialize(resp.GetType(), resp, out fsdata);
-        if (!r.Succeeded)
-            throw new WatsonException(r.FormattedMessages);
-
-        //  Convert fsdata to MessageResponse
-        MessageResponse messageResponse = new MessageResponse();
-        object obj = messageResponse;
-        r = _serializer.TryDeserialize(fsdata, obj.GetType(), ref obj);
-        if (!r.Succeeded)
-            throw new WatsonException(r.FormattedMessages);
+        Log.Debug("SpeechSandboxStreaming.OnMessage0()", "response: {0}", resp.Result.Output.Entities[0].Entity);
+        messageTested0 = true;
+    }
 
-        if (resp != null && (messageResponse.intents.Length > 0 || messageResponse.entities.Length > 0))
+    void OnMessage(DetailedResponse<MessageResponse> resp, IBMError error)
+    {
+        if (resp != null && resp.Result.Output.Intents.Count  != 0 )
         {
-            string intent = messageResponse.intents[0].intent;
+            string intent = resp.Result.Output.Intents[0].Intent;
             Debug.Log("Intent: " + intent);
             string currentMat = null;
             string currentScale = null;
             string direction = null;
             if (intent == "move")
             {
-                foreach (RuntimeEntity entity in messageResponse.entities)
+                foreach (RuntimeEntity entity in resp.Result.Output.Entities)
                 {
-                    Debug.Log("entityType: " + entity.entity + " , value: " + entity.value);
-                    direction = entity.value;
+                    Debug.Log("entityType: " + entity.Entity + " , value: " + entity.Value);
+                    direction = entity.Value;
                     gameManager.MoveObject(direction);
                 }
             }
 
             if (intent == "create")
             {
                 bool createdObject = false;
-                foreach (RuntimeEntity entity in messageResponse.entities)
+                foreach (RuntimeEntity entity in resp.Result.Output.Entities)
                 {
-                    Debug.Log("entityType: " + entity.entity + " , value: " + entity.value);
-                    if (entity.entity == "material")
-                    {
-                        currentMat = entity.value;
+                    Debug.Log("entityType: " + entity.Entity + " , value: " + entity.Value);
+                    if (entity.Entity == "material")
+                    { 
```

Comment:
there's a whitespace and indent issue here

Reply:
I don't see the issue.